.github/workflows/scheduled-full-ci.yml

```diff
@@ -29,6 +29,7 @@ jobs:
       - name: Run unit tests and generate report
         if: always()
         run: |
+          pip install -e .
```

Comment:
minor, but is `-e` really necessary? I think that's mainly when you want keep developing code in that package. the runner just needs to install and use.

Reply:
Didn't really think this, I just copied it over. I suppose removing it is better, although I suspect minor difference in running the pipelines.